pyrenew/process/ar.py

```diff
@@ -52,60 +48,113 @@ def sample(
         init_vals : ArrayLike
             Array of initial values. Must have the
             same first dimension size as the order.
-        noise_sd : float | ArrayLike
-            Scalar giving the s.d. of the AR
+        noise_sd : ArrayLike
+            Standard deviation of the AR
             process Normal noise, which by
             definition has mean 0.
-        **kwargs : dict, optional
-            Additional keyword arguments passed to
-            self.noise_rv_.sample()
 
         Returns
         -------
         ArrayLike
+            of shape (n,) + init_vals.shape[1:].
         """
-        noise_sd_arr = jnp.atleast_1d(noise_sd)
-        if not noise_sd_arr.shape == (1,):
-            raise ValueError("noise_sd must be a scalar. " f"Got {noise_sd}")
         autoreg = jnp.atleast_1d(autoreg)
+        noise_sd = jnp.array(noise_sd)
```

Comment:
Why are these not all `jnp.atleast_1d` or all `jnp.array`?

Reply:
Thinking a bit more.

`autoreg` can be a vector for each `plate` element, while `noise_sd` should be scalar for each `plate` element, so I understand there might be a difference for how they are handled. Not grasping why this is the difference, though.

Reply:
`noise_sd` can be a scalar, but the other two need a first dimension (time), even if it's length 1, because that first dimension length gives the order of the process.

Reply:
clarifying comment added to code in 3023a3d

Reply:
added a clarifying comment on this point to the code in 3023a3d